Don't try to set gcc/g++ flags for non-GNU compilers.

Submitted upstream at:
https://sourceforge.net/tracker/?func=detail&atid=517001&aid=2767126&group_id=67131

diff -urp -x '*~' -x '*.orig' configure.in configure.in
--- configure.in	2009-04-15 16:27:40.888792000 -0700
+++ configure.in	2009-04-15 16:27:43.635906000 -0700
@@ -9,19 +9,27 @@ AC_PROG_AWK
 AC_PROG_CC
 AC_GNU_SOURCE
 
-CFLAGS=-O2
+if test "x$GCC" = xyes; then
+    CFLAGS="-O2"
+fi
 AC_ARG_ENABLE([debug], [AC_HELP_STRING([--enable-debug],
     [build with debugging information default=no])],
     [test "$enableval" = "yes" && CFLAGS="-g"])
-CFLAGS="$CFLAGS -Wall"
+if test "x$GCC" = xyes; then
+    CFLAGS="$CFLAGS -Wall"
+fi
 
 AC_PROG_CXX
 
-CXXFLAGS=-O2
+if test "x$GXX" = xyes; then
+    CXXFLAGS="-O2"
+fi
 AC_ARG_ENABLE([debug], [AC_HELP_STRING([--enable-debug],
     [build with debugging information default=no])],
     [test "$enableval" = "yes" && CXXFLAGS="-g"])
-CXXFLAGS="$CXXFLAGS -Wall"
+if test "x$GXX" = xyes; then
+    CXXFLAGS="$CXXFLAGS -Wall"
+fi
 
 AC_PROG_INSTALL
 AM_DISABLE_SHARED
